pkg/plugins/gateway/algorithms/prefix_cache.go

```diff
@@ -72,10 +73,7 @@ func (p prefixCacheRouter) Route(ctx context.Context, pods map[string]*v1.Pod, m
 		}
 	}
 
-	tokens, err := utils.TokenizeInputText(message)
-	if err != nil {
-		return "", err
-	}
+	tokens := strings.Split(message, " ")
```

Comment:
Can we have some interface here so we can probably configure it for differenc scenarios? My concern is  I think it's ok to split by `" "` for english,, for some other languages like chinese, this might not be helpful.

Reply:
yeah, character way works.

Reply:
@varungup90 could you help rebase the change and address the comments? rune is int32 which is used to present unicode so it can represent almost any languages. using split way seems problematic for some languages.  WDYT?

Reply:
I am almost done with adding the interface for string based along with current token based.

I am thinking to speed up the merge, we can keep string based as-is and add another character based one. Only concern with character based is that it will generate too many tokens. No harm is providing it as option and user can choose.

Reply:
sounds good. Please resolve the conflicts and let's get this one merged.